sap-vcap-services/src/main/java/com/sap/cloud/environment/servicebinding/SapVcapServicesServiceBindingAccessor.java

```diff
@@ -33,40 +33,72 @@ public class SapVcapServicesServiceBindingAccessor implements ServiceBindingAcce
     @Nonnull
     private static final Logger logger = LoggerFactory.getLogger(SapVcapServicesServiceBindingAccessor.class);
 
+    @Nonnull
+    private static final String VCAP_SERVICES = "VCAP_SERVICES";
+
+    @Nonnull
+    private static final String VCAP_SERVICES_FILE_PATH = "VCAP_SERVICES_FILE_PATH";
+
     /**
      * The default {@link Function} to read environment variables.
      */
     @Nonnull
     public static final Function<String, String> DEFAULT_ENVIRONMENT_VARIABLE_READER = System::getenv;
 
     @Nonnull
-    private static final String VCAP_SERVICES = "VCAP_SERVICES";
+    public static final Function<Path, String> DEFAULT_FILE_READER = path -> {
+        try {
+            return Files.readString(path);
+        }
+        catch( IOException e ) {
+            logger.error("Failed to read VCAP_SERVICES from file: {}", VCAP_SERVICES_FILE_PATH, e);
+            return null;
+        }
+    };
```

Comment:
Ideally, we would use:

```java
public static final Function<Path, String> DEFAULT_FILE_READER = Files::readString;
```

but unfortunately that isn't allowed by Java, since `readString` throws checked exceptions and functional interfaces can't deal with that.

Reply:
yes, this was my struggle, why I didn't add this in the first place but when I though a second time about his it makes the code better testable that is why I decided to refactor.